jena-arq/src/test/java/org/apache/jena/sparql/transaction/AbstractTestTransPromote.java

```diff
@@ -270,15 +281,30 @@ private void run_08(boolean b) {
     @Test
     public void promote_11() { promote_readCommit_txnCommit(true, false) ; }
     
-    @Test(expected = TDBTransactionException.class)
-    public void promote_12() { promote_readCommit_txnCommit(false, true) ; }
+    @Test
+    public void promote_12() { 
+        expect(()->promote_readCommit_txnCommit(false, true) ,
+               getTransactionExceptionClass()) ;
+    }
+    
+    private void expect(Runnable runnable, Class<?>...classes) {
```

Comment:
This might be a pinch clearer as `... Class<? extends Exception>...classes) {`...